src/Controller/EnMarche/MembershipController.php

```diff
@@ -189,14 +189,18 @@ public function completeAction(MembershipRegistrationProcess $membershipRegistra
      * @Entity("adherent", expr="repository.findOneByUuid(adherent_uuid)")
      * @Entity("activationToken", expr="repository.findByToken(activation_token)")
      */
-    public function activateAction(Adherent $adherent, AdherentActivationToken $activationToken, CallbackManager $callbackManager): Response
-    {
+    public function activateAction(
+        Adherent $adherent,
+        AdherentActivationToken $activationToken,
+        CallbackManager $callbackManager,
+        AdherentAccountActivationHandler $accountActivationHandler
+    ): Response {
         if ($this->getUser()) {
             $this->redirectToRoute('app_search_events');
         }
 
         try {
-            $this->get('app.adherent_account_activation_handler')->handle($adherent, $activationToken);
+            $accountActivationHandler->handle($adherent, $activationToken);
```

Comment:
I am wondering why are you injecting `AdherentAccountActivationHandler` here but not `MembershipRequestHandler::class` 3 lines above ?

Reply:
I have modified `AdherentAccountActivationHandler` service by adding a new argument and I make it private. As this service is now private, I change **only** its injections and don't touch `MembershipRequestHandler` service for staying in the scope of my card.